adapters.go

```diff
@@ -152,8 +154,10 @@ func (a *whisperAdapter) decodeMessage(message *whisper.Message) (*protocol.Stat
 	if err != nil {
 		return nil, err
 	}
-	decoded.ID = message.Hash
-	decoded.SigPubKey = publicKey
+	// Don't override signature if present
```

Comment:
what do you mean exactly?

Reply:
In this case, I would add something like this:
```suggestion
	// Getting a signature from the Whisper message should happen only if
	// the signature was not defined in the payload itself.
```

And we can move `publicKey, err := crypto.UnmarshalPubkey(message.Sig)` inside `if decoded.SigPubKey == nil` to avoid unnecessary work in case of wrapped messages.

Reply:
makes sense